src/control/server/storage/bdev.go

```diff
@@ -24,6 +31,61 @@ import (
 // BdevPciAddrSep defines the separator used between PCI addresses in string lists.
 const BdevPciAddrSep = " "
 
+// BioState represents the health state of BIO device on an NVMe device (a blobstore on a NVMe
+// namespace).
+type BioState uint32
+
+// BioState constant definitions to represent expected bitset flag combinations.
+const (
+	BioStateNew      BioState = C.NVME_DEV_FL_PLUGGED
+	BioStateNormal   BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE
+	BioStateFaulty   BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE | C.NVME_DEV_FL_FAULTY
+	BioStateIdentify BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE | C.NVME_DEV_FL_IDENTIFY
+)
+
+// IsNew returns true if SSD is not in use by DAOS.
+func (bs BioState) IsNew() bool {
+	return bs&C.NVME_DEV_FL_INUSE == 0
+}
+
+// IsNormal returns true if SSD is in a normal, non-faulty state.
+func (bs BioState) IsNormal() bool {
+	return (bs&C.NVME_DEV_FL_PLUGGED != 0 && bs&C.NVME_DEV_FL_INUSE != 0 &&
+		bs&C.NVME_DEV_FL_FAULTY == 0)
+}
+
+// IsFaulty returns true if SSD is in a faulty state.
+func (bs BioState) IsFaulty() bool {
+	return bs&C.NVME_DEV_FL_FAULTY != 0
+}
+
+func (bs BioState) String() string {
```

Comment:
Now that I properly understand what this type represents, I think that maybe this stringer should probably be replaced with the guts of the `States()` method, no? As it is currently, the stringer is trying to boil down a bitmask into a single state string, but this won't always be correct, will it?

Instead of using the stringer for this, maybe what you want is a `StatusString()` method which correctly summarizes the device status for a human based on the various combinations of bitmask values.